ReactiveCocoa/Swift/Property.swift

```diff
@@ -90,16 +90,13 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	public var value: Value {
 		get {
 			lock.lock()
-			let value = _value
-			lock.unlock()
-			return value
+			defer { lock.unlock() }
+
+			return _value
```

Comment:
I'd consider using `withValue` here if we want to remove the duplication and guarantee that reads and writes always go through the same (locking) path: `withValue` and `modify`. What do you think?

This could be `return self.withValue { $0 }`